src/plugins/discover/public/context_awareness/profile_providers/register_profile_providers.test.ts

```diff
@@ -12,36 +12,22 @@ import { exampleDataSourceProfileProvider } from './example_data_source_profile'
 import { exampleDocumentProfileProvider } from './example_document_profile';
 import { exampleRootProfileProvider } from './example_root_pofile';
 import {
-  registerEnabledProfileProviders,
   registerProfileProviders,
+  registerProfileProvidersInternal,
 } from './register_profile_providers';
 
-describe('registerEnabledProfileProviders', () => {
-  it('should register enabled profile providers', async () => {
+describe('registerProfileProvidersInternal', () => {
+  it('should register all profile providers', async () => {
     const { rootProfileServiceMock, rootProfileProviderMock } = createContextAwarenessMocks({
       shouldRegisterProviders: false,
     });
-    registerEnabledProfileProviders({
+    registerProfileProvidersInternal({
       profileService: rootProfileServiceMock,
       availableProviders: [rootProfileProviderMock],
-      enabledProfileIds: ['root-profile'],
     });
     const context = await rootProfileServiceMock.resolve({ solutionNavId: null });
     expect(rootProfileServiceMock.getProfile(context)).toBe(rootProfileProviderMock.profile);
   });
-
-  it('should not register disabled profile providers', async () => {
```

Comment:
If we move the filtering logic to `registerProfileProvidersInternal`, we should be able to restore an updated version of this test.

Reply:
Done in the same commit as mentioned above.